server/src/main/java/org/elasticsearch/search/fetch/subphase/StoredFieldsPhase.java

```diff
@@ -98,18 +89,12 @@ public void setNextReader(LeafReaderContext readerContext) {
             public void process(HitContext hitContext) {
                 Map<String, List<Object>> loadedFields = hitContext.loadedFields();
                 Map<String, DocumentField> docFields = new HashMap<>();
-                Map<String, DocumentField> metaFields = new HashMap<>();
                 for (StoredField storedField : storedFields) {
                     if (storedField.hasValue(loadedFields)) {
-                        DocumentField df = new DocumentField(storedField.name, storedField.process(loadedFields));
-                        if (storedField.isMetadataField) {
-                            metaFields.put(storedField.name, df);
-                        } else {
-                            docFields.put(storedField.name, df);
-                        }
+                        docFields.put(storedField.name, new DocumentField(storedField.name, storedField.process(loadedFields)));
                     }
                 }
-                hitContext.hit().addDocumentFields(docFields, metaFields);
+                hitContext.hit().addDocumentFields(docFields, Collections.emptyMap());
```

Comment:
use setDocumentField instead? we don't need the map at all at this point, you can simply call the method from within the loop for each field.